src/sentry/api/endpoints/auth_config.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import absolute_import
+
+from django.conf import settings
+from django.contrib.auth import REDIRECT_FIELD_NAME
+from django.core.urlresolvers import reverse
+from rest_framework.response import Response
+
+from sentry import newsletter
+from sentry.auth.superuser import is_active_superuser
+from sentry.constants import WARN_SESSION_EXPIRED
+from sentry.http import get_server_hostname
+from sentry.utils import auth
+from sentry.models import Organization
+from sentry.api.base import Endpoint
+from sentry.web.frontend.auth_login import additional_context
+from sentry.web.frontend.base import OrganizationMixin
+
+
+class AuthConfigEndpoint(Endpoint, OrganizationMixin):
+    # Disable authentication and permission requirements.
+    permission_classes = []
+
+    def get(self, request, *args, **kwargs):
+        """
+        Get context required to show a login page. Registration is handled elsewhere.
+        """
+        if request.user.is_authenticated():
+            # if the user is a superuser, but not 'superuser authenticated' we
+            # allow them to re-authenticate to gain superuser status
+            if not request.user.is_superuser or is_active_superuser(request):
+                return self.respond_authenticated(request)
```

Comment:
This is a code path I was not able to test during hackweek, might be worth a second look at it.

Reply:
It's tested in the `test_superuser_is_not_redirected` test.